projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php

```diff
@@ -160,9 +160,14 @@ public function is_task_list_enabled( $id ) {
 	 */
 	public function is_task_list_dismissed( $id ) {
 		$task_list_dismissed_status = $this->get_task_list_dismissed_status();
+		$task_list_dismissed_until  = $this->get_task_list_dismissed_until();
+		$is_dismissed               = isset( $task_list_dismissed_status[ $id ] ) && true === $task_list_dismissed_status[ $id ];
 
+		if ( ! $is_dismissed && isset( $task_list_dismissed_until ) ) {
+			return new DateTime( 'now', new DateTimeZone( 'UTC' ) ) <= new DateTime( '@' . $task_list_dismissed_until, new DateTimeZone( 'UTC' ) );
```

Comment:
Maybe it would be more clear if we attribute to the variable each part of this code:
```suggestion
// Current UTC datetime
$current_time = new DateTime('now', new DateTimeZone('UTC'));

// Task list dismissed until datetime
$dismissed_until = new DateTime('@' . $task_list_dismissed_until, new DateTimeZone('UTC'));

// Check if current time is before or equal to the dismissed until time
return $current_time <= $dismissed_until;
```

Reply:
Thanks!
I improved the code based on your suggestion here d8c956db524bd35e1933fce809d4468d21eff406